app/controllers/home_controller.rb

```diff
@@ -1,5 +1,6 @@
 class HomeController < ApplicationController
   before_action :require_user, only: %i(subscriptions nearby)
+  before_action :check_email_verified, only: %i(dashboard)
```

Comment:
I guess, the notice will be shown at the dashboard only, so let's add a button somewhere on Profile page or maybe settings page to allow users to voluntarily go there and there they can "Request Confirmation Email" otherwise instead of button show a green tick with text as "Email Verified". How does this sound?

Reply:
I think this is a better idea. A button would be more user friendly. I ll add the button on the profiles page?

> Great work @shubhscoder  ! Just left some ideas, let me know what you think. Thanks!

Thank you so much ! I will the make the changes as soon as possible then we can probably merge this.

Reply:
Great!